cmd/policy-assistant/pkg/cli/analyze.go

```diff
@@ -296,59 +308,291 @@ func shouldIncludeANPandBANP(client *kubernetes.Clientset) (bool, bool) {
+							Namespace:       traffic.Destination.Internal.Namespace,
+						},
+						IP: traffic.Destination.IP,
+					}
+					allTrafficTmp := []*matcher.Traffic{
+						{
+							Source:       podA,
+							Destination:  podB,
+							ResolvedPort: traffic.ResolvedPort,
+							Protocol:     traffic.Protocol,
+						},
+					}
+					allTraffic = append(allTraffic, allTrafficTmp[0])
+				} else if traffic.Source.Internal.Workload == "" && traffic.Destination.Internal.Workload != "" {
+					destinationWorkloadInfo = matcher.WorkloadStringToTrafficPeer(traffic.Destination.Internal.Workload)
+					podB := &matcher.TrafficPeer{
+						Internal: &matcher.InternalPeer{
+							PodLabels:       destinationWorkloadInfo.Internal.PodLabels,
+							NamespaceLabels: destinationWorkloadInfo.Internal.NamespaceLabels,
+							Namespace:       destinationWorkloadInfo.Internal.Namespace,
+							Workload:        destinationWorkloadInfo.Internal.Workload,
+						},
+						IP: destinationWorkloadInfo.Internal.Pods[0].IP,
+					}
+					podA := &matcher.TrafficPeer{
+						Internal: &matcher.InternalPeer{
+							PodLabels:       traffic.Source.Internal.PodLabels,
+							NamespaceLabels: traffic.Source.Internal.NamespaceLabels,
+							Namespace:       traffic.Source.Internal.Namespace,
+						},
+						IP: traffic.Source.IP,
+					}
+					allTrafficTmp := []*matcher.Traffic{
+						{
+							Source:       podA,
+							Destination:  podB,
+							ResolvedPort: traffic.ResolvedPort,
+							Protocol:     traffic.Protocol,
+						},
+					}
+					allTraffic = append(allTraffic, allTrafficTmp[0])
+				} else {
+					podA := &matcher.TrafficPeer{
+						Internal: &matcher.InternalPeer{
+							PodLabels:       traffic.Source.Internal.PodLabels,
+							NamespaceLabels: traffic.Source.Internal.NamespaceLabels,
+							Namespace:       traffic.Source.Internal.Namespace,
+						},
+						IP: traffic.Source.IP,
+					}
+					podB := &matcher.TrafficPeer{
+						Internal: &matcher.InternalPeer{
+							PodLabels:       traffic.Destination.Internal.PodLabels,
+							NamespaceLabels: traffic.Destination.Internal.NamespaceLabels,
+							Namespace:       traffic.Destination.Internal.Namespace,
+						},
+						IP: traffic.Destination.IP,
+					}
+					allTrafficTmp := []*matcher.Traffic{
+						{
+							Source:       podA,
+							Destination:  podB,
+							ResolvedPort: traffic.ResolvedPort,
+							Protocol:     traffic.Protocol,
+						},
+					}
+					allTraffic = append(allTraffic, allTrafficTmp[0])
+				}
+			}
+		}
+	} else {
+
+		if protocol != "TCP" && protocol != "UDP" && protocol != "SCTP" {
+			logrus.Fatalf("Bad Protocol Value: protocols supported are TCP, UDP and SCTP")
+		}
+
+		sourceWorkloadInfo = matcher.WorkloadStringToTrafficPeer(sourceWorkloadTraffic)
+		destinationWorkloadInfo = matcher.WorkloadStringToTrafficPeer(destinationWorkloadTraffic)
+
+		if sourceWorkloadInfo.Internal.Pods == nil || destinationWorkloadInfo.Internal.Pods == nil {
+			return
```

Comment:
could we `logrus.Warn()` here that we couldn't find source workload or couldn't find dst workload?

Reply:
This is allready handled on the translate func, if the workload doesnt exist or if it exists but is scaled down to 0 i will display this : 
![image](https://github.com/user-attachments/assets/01885f94-9830-4b0a-b9a4-3badf846265a)